### PR TITLE
Employee firing

### LINKED ISSUE #46: employee firing
This possibly should be implemented at meetings by raycasting employee that's designated to be fired

### PR DESCRIPTION
**Pull request by mertwole**
Resolves #46

### CODE REVIEW

File: JamGame/Assets/Scripts/Location/EmployeeManager/Controller.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            firingMode ^= true;
+        }
+
+        public void Update()
+        {
+            if (firingMode && inputActions.UI.LeftClick.WasPressedThisFrame())
+            {
+                TryFireEmployee();
+            }
+        }
```
Comment (Kiuh) [inline JamGame/Assets/Scripts/Location/EmployeeManager/Controller.cs:48]: Make it on Awake like `inputActions.UI.LeftClick.performed += ...`

File: JamGame/Assets/Scripts/Location/EmployeeManager/Model.cs [hunk 2]
```diff
@@ -9,23 +9,27 @@
...
+    [AddComponentMenu("Scripts/Location/EmployeeManager/Location.EmployeeManager.Model")]
+    public class Model : MonoBehaviour
     {
         private DataProvider<EmployeeAmount> employeeAmountDataProvider;
         private DataProvider<MaxStress> maxStressDataProvider;
         private DataProvider<AllEmployeesAtMeeting> allEmployeesAtMeetingDataProvider;
         private DataProvider<AllEmployeesAtHome> allEmployeesAtHomeDataProvider;
 
+        [SerializeField]
+        private UnityEvent<EmployeeImpl> employeeFired;
```
Comment (Kiuh) [inline JamGame/Assets/Scripts/Location/EmployeeManager/Model.cs:31]:
```suggestion
        public event Action<EmployeeImpl> EmployeeFired;
```

**Review by Kiuh — CHANGES_REQUESTED** [review]
Make that all UI block firing and impossibility end meeting without ending firing mode.

And delete empty `NeedProvider` folder in `Scripts/Location` and correct namespace in `JamGame/Assets/Scripts/Employee/StressMeter/Inspector.cs`

File: JamGame/Assets/Scripts/Location/EmployeeManager/Model.cs (hunk 2 above)
Comment (mertwole) [reply JamGame/Assets/Scripts/Location/EmployeeManager/Model.cs:31]: It will create unnecessary dependency

Comment (mertwole) [reply JamGame/Assets/Scripts/Location/EmployeeManager/Model.cs:31]: discussed

File: JamGame/Assets/Scripts/Location/EmployeeManager/Controller.cs [hunk 3]
```diff
@@ -51,6 +58,13 @@ public void FiringButtonPressed()
         private void TryFireEmployee()
         {
             Vector3 position = inputActions.UI.Point.ReadValue<Vector2>();
+
+            IEnumerable<GameObject> ui_hits = RayCaster.UIRayCast(position);
+            if (ui_hits.Any(hit => ((1 << hit.layer) & uiMask) != 0))
```
Comment (Kiuh) [inline JamGame/Assets/Scripts/Location/EmployeeManager/Controller.cs:63]: Make this lambda in separated method.

**Review by Kiuh — APPROVED** [review]

**Pull request merged**